### PR TITLE
executor: reuse memory for parallelSortSpillHelper

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/53861

Problem Summary:

### What changed and how does it work?

we get a profiler while we benchmark sync load. it takes too much time on the newParallelSortSpillHelper. it is unnecessary.

https://flamegraph.com/share/4dca9137-2419-11ef-ab1b-b6898751f631

image

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)

after optimization

t takes less time on the newParallelSortSpillHelper

https://flamegraph.com/share/822c3df8-247d-11ef-ab1b-b6898751f631

image

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by xzhangxian1008 — APPROVED** [review]
LGTM

File: pkg/executor/sortexec/parallel_sort_spill_helper.go [hunk 1]
```diff
@@ -52,14 +52,15 @@ func newParallelSortSpillHelper(sortExec *SortExec, fieldTypes []*types.FieldTyp
...
-		tmpSpillChunk: chunk.NewChunkWithCapacity(fieldTypes, spillChunkSize),
+		tmpSpillChunk: chunk.NewChunkFromPoolWithCapacity(fieldTypes, spillChunkSize),
 	}
 }
 
 func (p *parallelSortSpillHelper) close() {
 	for _, inDisk := range p.sortedRowsInDisk {
 		inDisk.Close()
 	}
+	p.tmpSpillChunk.Destroy(spillChunkSize, p.fieldTypes)
```
Comment (windtalker) [inline pkg/executor/sortexec/parallel_sort_spill_helper.go:63]: put it in defer block?

Comment (hawkingrei) [reply pkg/executor/sortexec/parallel_sort_spill_helper.go:63]: I think it is impossible to have a problem with the ```isDisk.Close``` and lead to panic  here.

Comment (windtalker) [reply pkg/executor/sortexec/parallel_sort_spill_helper.go:63]: OK

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**